### ISSUE TITLE
BatchNormalization layers in Tensorflow 2.4.1 give constant validation accuracy

### ISSUE THREAD

**Issue by TianruiZhang** [tensorflow/tensorflow#48837]
<em>Please make sure that this is a bug. As per our
[GitHub Policy](https://github.com/tensorflow/tensorflow/blob/master/ISSUES.md),
we only address code/doc bugs, performance issues, feature requests and
build/installation issues on GitHub. tag:bug_template</em>

**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow): No
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Ubuntu 20.04.2 LTS
- Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device: N/A
- TensorFlow installed from (source or binary): Anaconda (conda 4.10.1 using conda install)
- TensorFlow version (use command below): 2.4.1
- Python version: Python 3.8
- Bazel version (if compiling from source): N/A
- GCC/Compiler version (if compiling from source): N/A
- CUDA/cuDNN version: 11.2
- GPU model and memory: GeForce GTX 1080, 7845MB

You can collect some of this information using our environment capture
[script](https://github.com/tensorflow/tensorflow/tree/master/tools/tf_env_collect.sh)
You can also obtain the TensorFlow version with:
1. TF 1.0: `python -c "import tensorflow as tf; print(tf.GIT_VERSION, tf.VERSION)"`
2. TF 2.0: `python -c "import tensorflow as tf; print(tf.version.GIT_VERSION, tf.version.VERSION)"`

**Describe the current behavior**

I was building a simple CNN for a binary image classification task and found something confusing regarding the use of the `BatchNormalization` layer in Tensorflow. There are 320 images in the training set with evenly divided negative and positive cases. There are 80 images in the validation set with evenly divided negative and positive cases. I set the `batch_size` for both training and validation set to 32.

Here is the architecture of my original model.
```
from tensorflow.keras import Sequential
from tensorflow.keras.layers import Input, Conv2D, MaxPooling2D, Flatten, Dropout, Dense
        
model = Sequential([
    Input(shape=(256, 256, 3]),
    Conv2D(64, (3, 3), padding=“same”, activation=“relu”),
    Conv2D(64, (3, 3), padding=“same”, activation=“relu”),
    MaxPooling2D((2, 2)),
    Conv2D(128, (3, 3), padding=“same”, activation=“relu”),
    Conv2D(128, (3, 3), padding=“same”, activation=“relu”),
    MaxPooling2D((2, 2)),
    Conv2D(256, (3, 3), padding=“same”, activation=“relu”),
    Conv2D(256, (3, 3), padding=“same”, activation=“relu”),
    Conv2D(256, (3, 3), padding=“same”, activation=“relu”),
    MaxPooling2D((2, 2)),
    Flatten(),
    Dense(128, activation=“relu”),
    Dropout(0.5),
    Dense(64, activation=“relu”),
    Dense(1, activation=“sigmoid”)
])

```
Both training and validation accuracies increased from somewhere between 0.5 and 0.6 and eventually converged somewhere between 0.95, which is good. However, with everything else unchanged, if I introduced `BatchNormalization` layers between `Conv2D` layers like the following:
```

model = Sequential([
    Input(shape=(256, 256, 3]),
    Conv2D(64, (3, 3), padding=“same”),
    BatchNormalization(),
    ReLU(),
    Conv2D(64, (3, 3), padding=“same”, activation=“relu”),
    MaxPooling2D((2, 2)),
    Conv2D(128, (3, 3), padding=“same”),
    BatchNormalization(),
    ReLU(),
    Conv2D(128, (3, 3), padding=“same”, activation=“relu”),
    MaxPooling2D((2, 2)),
    Conv2D(256, (3, 3), padding=“same”),
    BatchNormalization(),
    ReLU(),
    Conv2D(256, (3, 3), padding=“same”),
    BatchNormalization(),
    ReLU(),
    Conv2D(256, (3, 3), padding=“same”, activation=“relu”),
    MaxPooling2D((2, 2)),
    Flatten(),
    Dense(128, activation=“relu”),
    Dropout(0.5),
    Dense(64, activation=“relu”),
    Dense(1, activation=“sigmoid”)
])

```
The training accuracy still behaved the same way as before but validation accuracy remained constant at 0.5. I also tried several other implementations of `BatchNormalization` layers and had the same observation.
```

model = Sequential([
    Input(shape=(256, 256, 3]),
    Conv2D(64, (3, 3), padding=“same”, activation=“relu”),
    BatchNormalization(),
    Conv2D(64, (3, 3), padding=“same”, activation=“relu”),
    MaxPooling2D((2, 2)),
    Conv2D(128, (3, 3), padding=“same”, activation=“relu”),
    BatchNormalization(),
    Conv2D(128, (3, 3), padding=“same”, activation=“relu”),
    MaxPooling2D((2, 2)),
    Conv2D(256, (3, 3), padding=“same”, activation=“relu”),
    BatchNormalization(),
    Conv2D(256, (3, 3), padding=“same”, activation=“relu”),
    BatchNormalization(),
    Conv2D(256, (3, 3), padding=“same”, activation=“relu”),
    MaxPooling2D((2, 2)),
    Flatten(),
    Dense(128, activation=“relu”),
    Dropout(0.5),
    Dense(64, activation=“relu”),
    Dense(1, activation=“sigmoid”)
])

```
There are people online saying that there are bugs in Keras’s implementation of `BatchNormalization`. I had tried to reduce the layers of `BatchNormalization` and found that with even a single layer of `BatchNormalization` introduced, the validation accuracy will be constant at 0.5. Do you have any ideas why?

**Describe the expected behavior**

 I am rather new with deep learning and from what I have read recently on different tutorials about `BatchNormalization`, it should usually (though not always) be a performance booster. However, from what I have observed, `BatchNormalization` was ruining my validation performance. 

**Standalone code to reproduce the issue**
Provide a reproducible test case that is the bare minimum necessary to generate
the problem. If possible, please share a link to Colab/Jupyter/any notebook.

**Other info / logs** Include any logs or source code that would be helpful to
diagnose the problem. If including tracebacks, please include the full
traceback. Large logs and files should be attached.

**Comment by Saduf2019** [comment]
@TianruiZhang 
Apologies for the delayed response, is this still an issue, please share a colab gist with the issue reported, can you try on tf 2.5 and let us know.
